update to create class to install puppetclassify gem

abuxton opened the pull request.

acidprime commented on manifests/puppetclassify/install.pp:
```diff
@@ -0,0 +1,13 @@
+class node_manager::puppetclassify::install(
+  $version = $node_manager::params::version
+) inherits node_manager::params {
+$gemprovider = $node_manager::params::gemprovider
```
any reason not to make this a class parameter , it would make this a little more independently flexible in the future.

abuxton replied:
Not really it was purely preferential, and based on an optimistic hope that the puppet_gem provider  will be a stable part of the code base. 
happy to refactor would you also include source parameter though as that was a consideration of exposing the provisioner also.

WhatsARanjit commented on Gemfile:
```diff
@@ -1,6 +1,29 @@
 gem 'puppetlabs_spec_helper', '0.10.3'
 gem 'webmock', '1.22.1'
-gem 'puppetclassify', '0.1.0'
+gem 'puppetclassify', '0.1.2'
```
Gemfile needs 0.1.3 as well.

gjngeldenhuis commented:
This would be really useful.

npwalker commented:
Is there something preventing this from being merged?  I'd like to use it myself.

WhatsARanjit commented:
We were updating the gem dependencies to the most currently releases tested.

@abuxton Can you update the Gemfile and I'll merge it in?

petems commented on spec/spec_helper.rb.old:
```diff
@@ -0,0 +1 @@
+require 'puppetlabs_spec_helper/module_spec_helper'
```
This needs removing

petems commented on manifests/init.pp:
```diff
@@ -0,0 +1,4 @@
+class node_manager{
+
+  require node_manager::puppetclassify::install
```
Wouldn't this force an upgrade if someone uses the module and is managing the gem elseware?

petems replied:
Probably safer to put it behind a $manage_puppetclassify_gem parameter or something?

petems commented:
@abuxton Can you squash the commits?

abuxton commented:
@petems Squashed, removed .old file and updated gem ref to 1.3, not refactoring for boolean usage, if you use the class it works, if you're managing it in some other manner you have no reason to use the class!

petems commented:
@abuxton Yep, I realised tha afterwards! :smile: I'll merge this after classes :+1:
